File: 9000-9999/9640 - Dino və putalar/solution.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    vector<pair<int,int>>u;
    vector<pair<int,int>>v;
    long long n, m, l, f, j, s = 0, i, x, y, r;
    cin >> n;
    f = n;
    j = n;
    m = n * 2;
    for (i = 0; i < m; i++){
        cin >> x >> y;
        r = y - x;
        u.push_back(make_pair(r, x));
        v.push_back(make_pair(r, y));
    }
    
    sort(u.begin(), u.end());
    sort(v.begin(), v.end());
    reverse(v.begin(), v.end());

    for (i = 0; i < m; i++) {
        if (f == 0)
            break;
        s = s + u[i].second;
        f--;
    }
    for (i = 0; i < m; i++) {
        if (j == 0)
            break;
        s = s + v[i].second;
        j--;
    }
    cout << s;
    return 0;
}

File: 9000-9999/9640 - Dino və putalar/solution.py
u = []
v = []
s = 0
n = int(input())
f = n
j = n
m = n*2
for i in range(m):
    x,y = map(int,input().split())
    r = y-x
    u.append([r,x])
    v.append([r,y])

u.sort(key=lambda x:x[0])
v.sort(key=lambda x:x[0], reverse=True)

for i in range(m):
    if f==0:
        break
    s = s + u[i][1]
    f-=1
for i in range(m):
    if j==0:
        break
    s = s + v[i][1]
    j-=1

print(s)